master/buildbot/process/properties.py

```diff
@@ -80,6 +82,36 @@ def master(self):
     def master(self, value):
         self._master = value
 
+    @property
+    def sourcestamps(self):
+        if self.build is not None:
+            return self.build.getAllSourceStamps()
```

Comment:
I don't like the idea of accessing a private attribute like this. Maybe we should tune the `asDict()` method so that it takes a `v2` parameter and returns the correct value accordingly?